### ISSUE TITLE
[question] Override source-folder for conan create?

### ISSUE THREAD

**Issue by drmacdon** [conan-io/conan#2291]
Is there a good example of a conan project which has the source in the same repository as the conanfile.py which does not export the sources into the conan cache?

By default, I would like my conanfile.py source() to clone and checkout a tag which matches the conanfile.py version so I can ensure it is the tag (and not potentially local uncommitted files copied into the conan cache).

This creates a problem validating before pushing/tagging the repository since commands like conan create will execute the source(). 

Currently, I manually call install, build, export-pkg, test to get a package in my conan cache built without calling the default source() which allows sanity checking prior to the commit.

It would be nice to be able to overload the source() with a source-folder command line option, but I'm really just looking for how others achieve and if I'm overlooking something.

**Comment by memsharded** [comment]
I understand your issue. Right now, the best approach would be what you are already doing, running the local-flow, then commit, then ``conan create``.

I also think that this is a legit potential enhancement, not just a question. I will label it and propose to have a look at it for 1.2. Not sure what will be possible to do, but lets have a look.

**Comment by Aalmann** [comment]
@memsharded 
I have seen the PR #2643 which could be a solution for this problem, but only for git, even the method is called _handle_cvs which implies more than one system. Btw. cvs itself is ambiguous and should be scm instead.
However, wouldn't it be better and easier and closer to this requirement to have an argument `-sf, --source-folder` at `conan create`?

Best Aalmann

**Comment by memsharded** [comment]
Thanks for the suggestion @Aalmann , good point, I will use ``scm`` instead.

The problem with just doing ``--source-folder`` for conan-create is that it breaks reproducibility, and we have very solid evidence that capturing the SCM information from the source folder is wanted in the majority of cases. So the approach using something like ``scm_source = "source"`` in the recipe can cover both use cases. It will automatically define source folder to be used for ``conan create``, but it will also try to capture the commit of the repo, so later ``conan install --build`` will also work.

**Comment by Aalmann** [comment]
You are right. But isn't this implementation/approach the same as described in #2371?

**Comment by tonka3000** [comment]
@memsharded Is that local-flow also used in the conan-build-tools? I'm actually having the same problem as @drmacdon
